### PR TITLE
use matches! where possible

### PR DESCRIPTION
**Pull request by adamnemecek**
**Connections**
_Link to the issues addressed by this PR, or dependent PRs in other repositories_

**Description**
_Describe what problem this is solving, and how it's solved._

**Testing**
_Explain how this change is tested._

**Checklist**

- [ ] Run `cargo fmt`.
- [ ] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [ ] Run `cargo xtask test` to run tests.
- [ ] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
Hey there! 👋🏻 Welcome back to the WGPU community. Thanks for taking the time to contribute!

<details>

<summary>There were some significant issues with this PR that I was, thankfully, able to correct with some <code>fixup!</code> commits. I'm going to put my notes on these into a <code>details</code> tag, since they're mostly narrative and not important for everybody to read.</summary>

Let's start by looking at the size of it:

Files changed: 128; 3891 additions, 3648 deletions; holy crap this is a BIG PR

Wow. That is a _huge_ change. What's in here? It 's difficult to tell, because there are a _ton_ of formatting changes, at first glance. I don't know what you're using to format the changes you've made, but it certainly isn't stable `rustfmt` on a default configuration! So, I checked out this PR locally with `gh pr checkout 5308`, so I could try to fix it myself. However, doing so yields this error:

```
cargo fmt
error: unexpected closing delimiter: `}`
   --> /Users/mozilla/workspace/mozilla/wgpu/naga/src/valid/mod.rs:273:1
    |
231 |     const fn is_sized(&self) -> bool {
    |                                      - this delimiter might not be properly closed...
...
244 |         }
    |         - ...as it matches this but it has different indentation
...
273 | }
    | ^ unexpected closing delimiter
```

…This code doesn't even compile, ack! And yet, `cargo fmt` _did_ simplify changes down to ~150 lines of change, which I committed as 43f320571. That's much more tractable to review.

It seems you've changed some places of the form `match expr { pat => true, _ => false }` to `match!(expr, pat)`. That seems reasonable. The simplification of the diff also made it easy to determine that you missed editing some statement boundaries in `TypeInner::is_sized`, which I was able to correct with f14e051a7. Now, things compile, and appear to be formatted properly.

</details>

Before we proceed further review with the fixes I've applied on top, let me make something abundantly clear:

> [!WARNING]
>
> In general, it is not reasonable to expect an untested non-draft PR with thousands of lines of changes to be reviewed if it was not discussed previously. Please take the time to do the following next time:
>
> 1. Ensure your changes actually compile. We can't merge if they don't.
> 1. Ensure your changes are formatted with stable `rustfmt` on default configuration, as requested in the PR template that you submitted.
> 1. Take the time to populate the fields of the PR template. These are important for helping us understand the intent and status of your PR.
> 1. Before you submit a change that can be measured in hundreds of lines or more, consider reaching out to the community before requesting review, so that your intent is aligned with WGPU's maintainership before you expend a lot of effort.

I recommend using a draft PR to address the above; our CI can help you figure out these issues before you actually request a review.

**Comment by adamnemecek** [comment]
wait this is wrong, let me try again.

**Comment by ErichDonGubler** [comment]
@adamnemecek: Waaait, I think I already fixed it! 😅
